### DISCUSSION TITLE
task-001-env

### DISCUSSION THREAD

**Discussion by fredchristianson** [fredchristianson/javascript-game-engine#10] [Announcements]
There is a [description of this task on youtube.](https://youtu.be/Fz49_r-siso)
The branch task-001-env exists.  
The file tasks/task-001-env.md has details.

Advanced implementation can also create a ResourceLoader module and load .json files with it.  We will want a way to load assets (json, js, CSS, and HTML) to insert dynamically.  

fetch is nice, but there are additional steps to turn the response into what is wanted.  a one-statement "await loadJSON()" will be useful.

**Comment by chylinski82** [comment]
Ok, i think I'm done, though myself I didn't have a clue how to go about it and without chatGPT, just using traditional online tools (Goggle, Stack Overflow, etc.) it would have probably took me a week. Now, how do I submit my branch for review? I tried `git push origin work-001-env-chris` and `git push origin user-chris-envmodule-proposal` but I don't have permissions to do that.

**Reply by fredchristianson** [reply]
I was thinking git push origin work-001-env-chris.
But I'm still learning this 🙂

**Reply by fredchristianson** [reply]
@chylinski82  Your collaborator status says "pending invite".  I think you need to accept before you have access.  I don't see a way to resend but I can remove you and add again if you don't have an invite.

**Reply by chylinski82** [reply]
Ah, sorry, haven't seen that. All right, branch `work-001-env-chris` submitted for review.
